### ISSUE TITLE
[Accessibility] Date picker ARIA role is not allowed

### ISSUE THREAD

**Issue by masivaia** [microsoft/fluentui#25864]
### Environment Information

- **Package version(s)**: (fill this out)  8.103.3, 7.201.0
- **Browser and OS versions**: (fill this out if relevant) Not relevant

### Describe the issue:
Sample of issue:
Path: button[aria-label="February"]
Snippet: <button role="gridcell" class="itemButton-454" aria-label="February" aria-selected="false" data-is-focusable="true" type="button" tabindex="-1">Feb</button>

For all the months, the ARIA role **gridcell** is not allowed for given element

#### Actual behavior:

ARIA role **gridcell** is set on the button element for each month.

#### Expected behavior:

ARIA role **gridcell** should be set on the parent element (td) not on the button as it is for each day button.

### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? (Yes, No)

Requested priority: (Blocking, High, Normal, Low) Normal

Products/sites affected: (if applicable) Proposal Workspace

**Comment by 1hw7** [comment]
Hi! We are experiencing the same issue. It would be great if the fluent team could change the role assigned to the attribute. Thanks for filing this @masivaia !

**Comment by WPoole** [comment]
Hey! Same issue here with no obvious workaround. This is a blocking issue for us and some help would be greatly appreciated. Even if there is a workaround while the team works on an actual fix that would be awesome!

**Comment by yatchiu** [comment]
+1
